src/AzureIdentityAccessTokenProvider.cs

```diff
@@ -44,29 +48,47 @@ public AzureIdentityAccessTokenProvider(TokenCredential credential, string [] al
 
         if(!_scopes.Any())
             _scopes.Add("https://graph.microsoft.com/.default"); //TODO: init from the request hostname instead so it doesn't block national clouds?
+
+        _obsOptions = observabilityOptions ?? new();
```

Comment:
I notice there are some inconsistencies with the approach made in https://github.com/microsoft/kiota-abstractions-dotnet/pull/34/files

- The `_activitySource` is a static member while here its lifetime is tied to the `AzureIdentityAccessTokenProvider` lifetime. ([point 1](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/distributed-tracing-instrumentation-walkthroughs#best-practices-1))
- In an app with multiple instances of `AzureIdentityAccessTokenProvider` there may be an issue with conflicting names in the source instances ([point 2](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/distributed-tracing-instrumentation-walkthroughs#best-practices-1))

Reply:
The reason it's done differently in abstractions is because we wanted the request content object type and didn't want to impact code generation for that information.
I don't think abstractions is going to receive more tracing, metrics or logging off any kind moving forward so this is why I thought it was ok to take a shortcut.
For authentication and http, this is just the first step and we needed to make sure things are configurable. Since the configuration is held by the provider, request adapter, the activity source needed to be instance specific and not static to avoid conflicts between multiple clients. I hope this makes sense.